### PR TITLE
Feat/e2e ci

### PR DESCRIPTION
**Pull request by JonathanConn**
## Changes
- feat: added CI for e2e tests

### Points of discussion
- Run on every push or PR? Each has its advantages, I am in favor of every push
- How long should we store the test reports? They could build up over time if we decide to run on every `push`

### Ideas for the future
- Instead of building each time, we could utilize cloudflare pages to test 

Important: This [web-example PR](https://github.com/WalletConnect/web-examples/pull/260) needs to be merged before theses tests will pass, as they test against our wallet web-example

### CODE REVIEW

File: .github/workflows/playwright_tests.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+env:
+  NEXT_PUBLIC_PROJECT_ID: ${{ secrets.NEXT_PUBLIC_PROJECT_ID }}
+
+jobs:
+  e2e_tests:
+      runs-on: ubuntu-latest
+      timeout-minutes: 30
+      permissions:
+        contents: read
+        deployments: write
```
Comment (arein) [inline .github/workflows/playwright_tests.yml:27]: curious: why do you need this?

**Review by arein — DISMISSED** [review]

File: .github/workflows/playwright_tests.yml (hunk 1 above)
Comment (JonathanConn) [reply .github/workflows/playwright_tests.yml:27]: Great catch, 
that was leftover from researching with cloudflare
It was needed so the GitHub cloudflare action I was using so use GITHUB_TOKEN with the cloudflare api
Will be removed 🤘

**Comment by JonathanConn** [comment]
Fixed some bugs in test that were passing locally, but failing in CI
Should be good to go now @arein

**Review by arein — DISMISSED** [review]
LGTM!

File: .github/workflows/playwright_tests.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
+name: Playwright Tests
+
+on: 
+  push:  
+    branches: '**'
```
Comment (arein) [inline .github/workflows/playwright_tests.yml:5]: Screenshot 2023-08-02 at 10 01 04 PM

I think this causes the tests to runs twice upon PR. Can you narrow down to `V2` and `V3`?

Comment (JonathanConn) [reply .github/workflows/playwright_tests.yml]: You're right, I actually had them swapped, fixed 👍

**Pull request merged**